core/src/test/scala/unit/kafka/server/ReplicaFetcherThreadTest.scala

```diff
@@ -89,9 +100,148 @@ class ReplicaFetcherThreadTest {
       t1p1 -> new EpochEndOffset(Errors.NONE, UNDEFINED_EPOCH, UNDEFINED_EPOCH_OFFSET)
     )
 
+    assertFalse("Leader request should not have been sent", toFail)
+    assertEquals("results from leader epoch request should have undefined offset", expected, result)
+  }
+
+  /**
+    * If a partition doesn't have an epoch in the cache, this means it either
+    *   does not support epochs (log message format below 0.11) or it is a bootstrapping follower.
+    *
+    * In both cases, he has an undefined epoch
+    *   and there is no use to send the request, as we know the broker will answer with that epoch
+    */
+  @Test
+  def shouldNotIssueLeaderEpochRequestIfIfBrokerProtocolAbove11AllTopicsDontSupportEpochs(): Unit = {
```

Comment:
The name seems a little stale. Perhaps `shouldNotSendEpochRequestIfLastEpochUndefinedForAllPartitions` or something like that?  The other test case probably needs a similar update.